### ISSUE TITLE
 compat is removed in version 2.0.0 - error with netbox inventory file

### ISSUE THREAD

**Issue by bsmeding** [ansible-collections/ansible.netcommon#232]
##### SUMMARY
When using latest collection ansible.netcommon with the netbox.netbox collection for dynamic inventory. I get a result that the compat module is missing. I see in Git that the folder indeed is missing between versions 1.5.0 and 2.0.0.

ansible_netwerk/netbox$ ansible-playbook update_all_info.yml -i ./inventories/netbox.yml
[WARNING]:  * Failed to parse /home/<<USER>>/ansible_netwerk/netbox/inventories/netbox.yml with auto plugin: No module named
'ansible_collections.ansible.netcommon.plugins.module_utils.compat'

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ansible.netcommon/plugins/module_utils/compat 
is removed

##### ANSIBLE VERSION

```
ansible 2.10.6
  config file = /home/bsmeding/ansible_netwerk/netbox/ansible.cfg
  configured module search path = ['/usr/local/share/virtualenv/ansible/lib/python3.7/site-packages/napalm_ansible/modules', '/var/lib/awx/venv/ansible/lib/python3.7/site-packages/napalm_ansible/modules', '/etc/ansible/modules/ntc-ansible/library', '/home/bsmeding/library', '/home/bsmeding/ansible_netwerk/netbox/library']
  ansible python module location = /data/virtualenv/ansible-2.10/lib/python3.9/site-packages/ansible
  executable location = /data/virtualenv/ansible-2.10/bin/ansible
  python version = 3.9.0 (default, Nov  9 2020, 11:59:41) [GCC 6.3.0 20170516]
bsmeding@mgmt-ansible1:~/ansible_netwerk/netbox$
```

##### CONFIGURATION
```
ANSIBLE_PIPELINING(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = -o ControlMaster=auto -o ControlPersist=60s -o PreferredAuthentications=publickey
ANSIBLE_SSH_CONTROL_PATH(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = ./ansible-%%r@%%h:%%p
ANSIBLE_SSH_RETRIES(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = 2
COLLECTIONS_PATHS(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = ['/home/bsmeding/ansible_netwerk/netbox/collections']
DEFAULT_ACTION_PLUGIN_PATH(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = ['/venv/ansible/lib/python3.7/']
DEFAULT_CALLBACK_WHITELIST(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = ['profile_tasks']
DEFAULT_FILTER_PLUGIN_PATH(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = ['/etc/ansible/modules/ntc-ansible/filter_plugins', '/home/bsmeding/ansible_netwerk/netbox/filter_plugins']
DEFAULT_FORKS(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = 20
DEFAULT_HOST_LIST(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = ['/home/bsmeding/ansible_netwerk/netbox/inventories/netbox_inventory_ftth_all.yml']
DEFAULT_MODULE_PATH(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = ['/usr/local/share/virtualenv/ansible/lib/python3.7/site-packages/napalm_ansible/modules', '/var/lib/awx/venv/ansible/lib/python3.7/site-
DEFAULT_POLL_INTERVAL(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = 15
DEFAULT_ROLES_PATH(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = ['/home/bsmeding/ansible_netwerk/netbox/roles']
DEFAULT_SCP_IF_SSH(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = True
DEFAULT_STDOUT_CALLBACK(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = skippy
DEFAULT_TIMEOUT(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = 120
DEFAULT_VAULT_PASSWORD_FILE(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = /home/bsmeding/ansible_netwerk/netbox/vault_pass.txt
DISPLAY_SKIPPED_HOSTS(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = False
HOST_KEY_CHECKING(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = False
PARAMIKO_LOOK_FOR_KEYS(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = False
PERSISTENT_COMMAND_TIMEOUT(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = 60
PERSISTENT_CONNECT_RETRY_TIMEOUT(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = 30
PERSISTENT_CONNECT_TIMEOUT(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = 120
RETRY_FILES_ENABLED(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = True
USE_PERSISTENT_CONNECTIONS(/home/bsmeding/ansible_netwerk/netbox/ansible.cfg) = True
```

##### OS / ENVIRONMENT
Debian 10

##### STEPS TO REPRODUCE

```yaml
bsmeding@mgmt-ansible1:~/ansible_netwerk/netbox$ cat inventories/netbox.yml
# netbox_inventory.yml file in YAML format
# Example command line: ansible-inventory -v --list -i netbox_inventory.yml

plugin: netbox.netbox.nb_inventory
api_endpoint: https://ansible-nw.network.intern
validate_certs: False
config_context: False
group_by:
  - device_roles
query_filters:
  - role: ftth_distribution
device_query_filters:
  - has_primary_ip: 'true'
```

##### EXPECTED RESULTS
```
bsmeding@mgmt-ansible1:~/ansible_netwerk/netbox$ ansible-inventory -v --list -i ./inventories/netbox.yml
Using /home/bsmeding/ansible_netwerk/netbox/ansible.cfg as config file
Fetching: https://ansible-nw.network.intern/api/docs/?format=openapi
Fetching: https://ansible-nw.network.intern/api/dcim/devices/?limit=0&role=ftth_distribution&has_primary_ip=true&exclude=config_context
Fetching: https://ansible-nw.network.intern/api/virtualization/virtual-machines/?limit=0&role=ftth_distribution&exclude=config_context
Fetching: https://ansible-nw.network.intern/api/dcim/sites/?limit=0
Fetching: https://ansible-nw.network.intern/api/dcim/regions/?limit=0
Fetching: https://ansible-nw.network.intern/api/tenancy/tenants/?limit=0
Fetching: https://ansible-nw.network.intern/api/dcim/racks/?limit=0
Fetching: https://ansible-nw.network.intern/api/dcim/rack-groups/?limit=0
Fetching: https://ansible-nw.network.intern/api/dcim/device-roles/?limit=0
Fetching: https://ansible-nw.network.intern/api/dcim/platforms/?limit=0
Fetching: https://ansible-nw.network.intern/api/dcim/device-types/?limit=0
Fetching: https://ansible-nw.network.intern/api/dcim/manufacturers/?limit=0
Fetching: https://ansible-nw.network.intern/api/virtualization/clusters/?limit=0
Fetching: https://ansible-nw.network.intern/api/ipam/services/?limit=0
{
    "_meta": {
        "hostvars": {
            "my-router1": {
                "ansible_host": "10.8.x.x",

---<OUTPUT COMMITET>
```

##### ACTUAL RESULTS

```
bsmeding@mgmt-ansible1:~/ansible_netwerk/netbox$ ansible-inventory -v --list -i ./inventories/netbox.yml
Using /home/bsmeding/ansible_netwerk/netbox/ansible.cfg as config file
[WARNING]: While constructing a mapping from /home/bsmeding/ansible_netwerk/netbox/inventories/netbox.yml, line 4, column 1, found a duplicate dict key (plugin). Using last defined value only.
[WARNING]:  * Failed to parse /home/bsmeding/ansible_netwerk/netbox/inventories/netbox.yml with auto plugin: No module named 'ansible_collections.ansible.netcommon.plugins.module_utils.compat'
[WARNING]:  * Failed to parse /home/bsmeding/ansible_netwerk/netbox/inventories/netbox.yml with yaml plugin: Plugin configuration YAML file, not YAML inventory
[WARNING]:  * Failed to parse /home/bsmeding/ansible_netwerk/netbox/inventories/netbox.yml with ini plugin: Invalid host pattern 'plugin:' supplied, ending in ':' is not allowed, this character is reserved to
provide a port.
[WARNING]: Unable to parse /home/bsmeding/ansible_netwerk/netbox/inventories/netbox.yml as an inventory source
[WARNING]: No inventory was parsed, only implicit localhost is available
{
    "_meta": {
        "hostvars": {}
    },
    "all": {
        "children": [
            "ungrouped"
        ]
    }
}
```

FIXED WITH ROLLING BACK VERSION
Because netcommon was back from 2.0.0 to 1.5.0  I also needed to rollback other version for compatibility
```
bsmeding@mgmt-ansible1:~/ansible_netwerk/netbox$ cat collections/requirements.yml
collections:
  - name: ansible.netcommon
    version: 1.5.0
  - name: netbox.netbox
    version: 1.2.1
  - name: cisco.ios
    version: 1.3.0
  - name: cisco.iosxr
    version: 1.2.1
bsmeding@mgmt-ansible1:~/ansible_net
```

**Comment by Qalthos** [comment]
This is correct, and an intentional, breaking change in version 2.0.0, as noted in our [changelog](https://github.com/ansible-collections/ansible.netcommon/blob/main/CHANGELOG.rst#breaking-changes-porting-guide).

This is not an issue with netcommon, and appears to be fixed in their [version 3.0.0](https://github.com/netbox-community/ansible_modules/blob/devel/CHANGELOG.rst#bugfixes).

**Comment by bsmeding** [comment]
> This is correct, and an intentional, breaking change in version 2.0.0, as noted in our [changelog](https://github.com/ansible-collections/ansible.netcommon/blob/main/CHANGELOG.rst#breaking-changes-porting-guide).
> 
> 
> 
> This is not an issue with netcommon, and appears to be fixed in their [version 3.0.0](https://github.com/netbox-community/ansible_modules/blob/devel/CHANGELOG.rst#bugfixes).

Didn't found it in a search, my bad. Thanks for the update
